.circleci/config.yml

```diff
@@ -0,0 +1,53 @@
+version: 2.1
+
+commands:
+  submodules:
+    name: "Pull submodules"
+    command: git submodule update --init
+  apt-dependencies:
+    name: "Install apt dependencies"
+    parameters:
+      llvm:
+        type: string
+    command: |
+      echo 'deb http://apt.llvm.org/stretch/ llvm-toolchain-stretch<< llvm >> main' | sudo tee /etc/apt/sources.list.d/llvm.list
+      wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key|sudo apt-key add -
+      sudo apt-get update
+      sudo apt-get install llvm python3 llvm<< llvm >>-dev clang<< llvm >> libclang<< llvm >>-dev gcc-arm-linux-gnueabi binutils-arm-none-eabi libc6-dev-armel-cross gcc-aarch64-linux-gnu libc6-dev-arm64-cross qemu-system-arm qemu-user gcc-avr avr-libc
+  dep:
+    name: "Install Go dependencies"
+    command: |
+      curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+      dep ensure --vendor-only
+
+jobs:
+  build:
+    docker:
+      - image: circleci/golang:1.11
+
+    working_directory: /go/src/github.com/tinygo-org/tinygo
+    steps:
+      - checkout
+      - submodules
+      - apt-dependencies:
+          llvm: "-7"
+      - dep
+      - run: go install .
+      - run: make test
```

Comment:
Should this not be in a separate `test` job?

Reply:
It seems to me that a new job would mean reinstalling all dependencies, which is rather expensive (at least until we have a special testing container).

Reply:
Fair point. We might be able to do the pulling in an initial job and then use WORKSPACE to populate for future jobs? https://circleci.com/docs/2.0/caching/.

Reply:
I don't think it's reasonably possible to cache dependencies installed by `apt-get`. What could be improved is to cache submodules or Go dependencies, but I'm not sure how to do that reliably. The Go build cache is already cached, which shaves of ~15s of build time. Improving the TinyGo build cache (for compiler-rt) would also save ~15s, but I'll need better caching logic for that (see buildcache.go).